src/main/scala/com.snowplowanalytics.lrumap/CreateLruMap.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2012-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.lrumap
+
+import cats.Eval
+import cats.syntax.functor._
+import cats.effect.Sync
+
+/** `CreateLruMap` provides an ability to initialize the cache,
+  * which effect will `F`
+  *
+  * *WARNING*: Due to support of non-`Sync`/eager algebras,
+  * users should be super-careful about creating *values* of `F[_]`,
+  * as they won't have RT/lazy semantics and will be memoized
+  *
+  * This is NOT a type class, as it does not have the coherence
+  * requirement, but can be passed implicitly
+  */
+trait CreateLruMap[F[_], K, V] {
+  /** Create an LruMap within `F` effect
+    *
+    * @param size Max size before evicting
+    */
+  def create(size: Int): F[LruMap[F, K, V]]
+}
+
+object CreateLruMap {
+
+  /** Summoner */
+  def apply[F[_], K, V](implicit ev: CreateLruMap[F, K, V]): CreateLruMap[F, K, V] = ev
+
+  // Based on com.twitter.util.LruMap
+  // https://github.com/twitter/util/blob/develop/util-collection/src/main/scala/com/twitter/util/LruMap.scala
+
+  /** Eager instance */
+  implicit def idInitCache[K, V]: CreateLruMap[Eval, K, V] = new CreateLruMap[Eval, K, V] {
```

Comment:
`evalInitCache`.

Also, I think it makes sense to leave all of them: `IO`, `Eval` and `Id` as well. Just `Id` is not something people should generally use.

Reply:
the problem is that supporting 3 `F` instead of 2 requires even more effort